third_party/blink/renderer/core/events/message_event.cc

```diff
@@ -265,6 +275,14 @@ void MessageEvent::initMessageEvent(
     MessagePortArray* ports,
     UserActivation* user_activation,
     mojom::blink::DelegatedCapability delegated_capability) {
+
+  recordreplay::Assert("[RUN-2037-2901] MessageEvent::initMessageEvent %d %d %zu %s %s",
+    source ? source->RecordReplayId() : -1,
```

Comment:
You only need this check here, not above?

Reply:
The other places are followed by:

```cpp
DCHECK(IsValidSource(source_.Get()));
```